src/datastore/postgres-store.ts

```diff
@@ -1099,18 +1150,17 @@ export class PgDataStore
     // TODO(mb): look up microblocks streamed off this block that where accepted by the next anchor block
     const microblocksStreamed: string[] = [];
 
-    // TODO(mb): replace `data.txs` with a list of all updated DbTx values (including orphaned microblock-txs, updated microblock-txs, batched-txs)
-    const txIdList = data.txs
-      .map(({ tx }) => ({ txId: tx.tx_id, txIndex: tx.tx_index }))
-      .sort((a, b) => a.txIndex - b.txIndex)
-      .map(tx => tx.txId);
-    this.emit('blockUpdate', data.block, txIdList, microblocksAccepted, microblocksStreamed);
-    data.txs.forEach(entry => {
-      this.emit('txUpdate', entry.tx);
+    await this.notifier.sendBlock({
+      blockHash: data.block.block_hash,
+      microblocksAccepted: microblocksAccepted,
+      microblocksStreamed: microblocksStreamed,
+    });
+    data.txs.forEach(async entry => {
+      await this.notifier.sendTx({ txId: entry.tx.tx_id });
```

Comment:
This needs to be a regular `for (...)` loop, otherwise `.forEach` will run all these concurrently, and not actually await. Unless that's the goal?